Client/Frontend/Browser/Favicons/FaviconFetcher.swift

```diff
@@ -237,8 +238,22 @@ class FaviconFetcher {
     }
     
     private func fetchIcon(_ completion: @escaping (URL, FaviconAttributes) -> Void) {
+        // Fetch icon logic:
+        // 1. Check if current domain has a favicon.
+        var correctFavicon = domain.favicon
+        
+        // 2. Private mode uses their own in-memory only favicons.
+        // If no in-memory favicon is found we look if there's any persisted one(from normal browsing mode)
+        // and use that one until in-memory favicon is saved.
+        if correctFavicon == nil,
+            PrivateBrowsingManager.shared.isPrivateBrowsing,
+            let urlString = domain.url,
+            let url = URL(string: urlString) {
```

Comment:
We can just give `Domain.getPersistedDomain` `self.url` instead of computing based on `domain` (this is what we use to make `domain` in `FaviconFetcher.init`)